### ISSUE TITLE
compilation c++ warning

### ISSUE THREAD

**Issue by AlexMielke1988** [AlexMielke1988/NetFACS#15]
when compiling, I get for combi_props.cpp:69:26, armadillo_functions.cpp:139:24, and armadillo_functions.cpp:140:30 the warning:
warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
for (size_t i = 0; i < inmat.ncol(); i++) {
                           ^

**Comment by gobbios** [comment]
Funny, this warning doesn't pop up for me. But I do remember the issue vaguely from some other project. Hopefully fixed now.

**Comment by gobbios** [comment]
Does this actually solve #14 too?

**Comment by AlexMielke1988** [comment]
I can use the parallelisation when I apply the function to data, but I can't have use_cpp and use_parallel TRUE as default when running the checks for CRAN. Beats me why

**Comment by gobbios** [comment]
Can you post the output log? There must be a temporary file somewhere that contains the complete log with all errors and warnings.

**Comment by gobbios** [comment]
Oh and one thing that you could try is to delete `Makevars.win` in the `src` folder, and run `devtools::check()` again. I seem to remember that this might have helped too. But don't ask me why.

**Comment by gobbios** [comment]
I managed to run `devtools::check()` successfully under Windows in a virtual machine, even with `use_cpp=TRUE` and `use_parallel=TRUE` in the examples. So it must be something in your toolchain, I guess. I had to tinker a bit with the Rtools to get it going, but I can't tell how much of this tinkering was due to me trying to run this in a VM. Anyway, I'll just uploaded a version to win-builder.r-project.org. Hopefully the report we get from there will enlighten us.

**Comment by gobbios** [comment]
Baby steps. I was partially successful with the win-builder. I changed the default in `netfacs()` to `n_cores = 2`. Not ideal, but it seems that in this way the examples don't crash the CRAN check, even with `use_cpp = TRUE`. Alternatively, we could just set all the examples to use 2 cores and leave the function definition to use all available cores minus 1, but for now I was too lazy to do that. Also, I added a line that turns off parallelisation entirely if there is only one core detected. 

And there are more issues remaining that circle around failures in `tests/`. But those don't seem to be related (directly at least) with neither the parallelisation nor the cpp issues. 

If this fix works for you, you can close this issue.

**Comment by AlexMielke1988** [comment]
For me, setting the settings to use_cpp = FALSE and use_parallel = TRUE work now, even though it seems to be slower than having parallel off for most functions, probably because I have two cores on my machine. I have changed the netfacs function to also set use_parallel = FALSE when there are two cores, because by our definition (n_cores = cores -1), that is essentially the same. 

For using use_cpp = TRUE, all functions throw out an error. I have attached the
[cpp error check.log](https://github.com/AlexMielke1988/NetFACS/files/4426252/cpp.error.check.log)
